### ISSUE TITLE
The `jf gradlec -repo-resolve` does not add extra repository into target Gradle build

### ISSUE THREAD

**Issue by artembilan** [jfrog/jfrog-cli#2364]
### Describe the bug

We have a GutHub Actions workflow to verify staged artifacts against downstream project before promoting release.
Something like this:
```
      - uses: jfrog/setup-jfrog-cli@v3
        env:
          JF_ENV_SPRING: ${{ secrets.JF_ARTIFACTORY_SPRING }}

      - name: Configure JFrog Cli
        run: jf gradlec --repo-resolve libs-staging-local

      - name: Verify Spring Integration Kafka against staged release
        run: |
          sed -i "1,/springKafkaVersion.*/s/springKafkaVersion.*/springKafkaVersion='${{ inputs.releaseVersion }}'/" build.gradle
          jf gradle :spring-integration-kafka:check
```

We have modeled this after our experience with Maven:
```
      - uses: jfrog/setup-jfrog-cli@v3
        env:
          JF_ENV_SPRING: ${{ secrets.JF_ARTIFACTORY_SPRING }}

      - name: Configure JFrog Cli
        run: jf mvnc --repo-resolve-releases=libs-release-staging --repo-resolve-snapshots=snapshot

      - name: Verify samples against staged release
        run: jf mvn verify -D"spring.amqp.version"=${{ inputs.releaseVersion }} -B -ntp
```

### Current behavior

According to the Maven CLI command implementation, it looks like there is an extra `buildInfoConfig.artifactoryResolutionEnabled` property propagated.
And therefore `ArtifactoryProjectBuilder` creates repositories configuration for Maven build tool: https://github.com/jfrog/build-info/blob/master/build-info-extractor-maven3/src/main/java/org/jfrog/build/extractor/maven/ArtifactoryProjectBuilder.java#L24.

We just don't see such an implementation for Gradle.
Here is a respective error we are facing:
```
Run jfrog/setup-jfrog-cli@v3
  with:
    version: 2.50.4
  env:
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.9-9/x64
    JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.9-9/x64
    JF_ENV_SPRING: ***
Setup JFrog CLI
  Downloading JFrog CLI from https://releases.jfrog.io/artifactory/jfrog-cli/v2/2.50.4/jfrog-cli-linux-amd64/jfrog
  /opt/hostedtoolcache/jf/2.50.4/x64/jf config import ***
  21:47:49 [Warn] You are using JFrog CLI version 2.50.4, however version 2.51.1 is available.
  To install the latest version, visit: https://jfrog.com/getcli
  To see the release notes, visit: https://github.com/jfrog/jfrog-cli/releases
  To avoid this message, set the JFROG_CLI_AVOID_NEW_VERSION_WARNING variable to TRUE
  21:47:49 [Info] Importing server ID 'repo.spring.io'
0s
Run jf gradlec --repo-resolve libs-staging-local
  jf gradlec --repo-resolve libs-staging-local
  shell: /usr/bin/bash -e {0}
  env:
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.9-9/x64
    JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.9-9/x64
    JFROG_CLI_ENV_EXCLUDE: *password*;*secret*;*key*;*token*;*auth*;JF_ARTIFACTORY_*;JF_ENV_*;JF_URL;JF_USER;JF_PASSWORD;JF_ACCESS_TOKEN
    JFROG_CLI_OFFER_CONFIG: false
    JFROG_CLI_BUILD_NAME: Verify Staged Artifacts
    JFROG_CLI_BUILD_NUMBER: 2
    JFROG_CLI_BUILD_URL: https://github.com/spring-projects/spring-kafka/actions/runs/7065535829
    JFROG_CLI_USER_AGENT: setup-jfrog-cli-github-action/3.4.1
21:47:49 [Info] gradle build config successfully created.
46s
Run sed -i "1,/springKafkaVersion.*/s/springKafkaVersion.*/springKafkaVersion='3.1.1'/" build.gradle
21:47:49 [Info] Running gradle...
21:47:49 [Info] Running gradle command: /usr/bin/gradle --version
21:47:52 [Info] Using Gradle version: 8.4
21:47:52 [Info] The build-info-extractor jar is not cached locally. Downloading it now...
You can set the repository from which this jar is downloaded.
Read more about it at https://jfrog.com/help/r/jfrog-cli/downloading-the-maven-and-gradle-extractor-jars
21:47:52 [Info] Downloading JFrog's Dependency from https://releases.jfrog.io/artifactory/oss-release-local/org/jfrog/buildinfo/build-info-extractor-gradle/5.1.5/build-info-extractor-gradle-5.1.5-uber.jar
21:47:52 [Info] Running gradle command: /usr/bin/gradle --init-script /home/runner/.jfrog/dependencies/gradle/5.1.5/gradle.init :spring-integration-kafka:check
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :spring-integration-test-support:updateCopyrights SKIPPED
> Task :spring-integration-core:updateCopyrights SKIPPED
> Task :spring-integration-kafka:updateCopyrights SKIPPED
> Task :spring-integration-kafka:processResources
> Task :spring-integration-kafka:processTestResources

> Task :spring-integration-core:compileKotlin FAILED
> Task :spring-integration-test-support:compileKotlin FAILED
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':spring-integration-test-support:compileKotlin'.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
> Failed to calculate the value of task ':spring-integration-test-support:compileKotlin' property 'scriptExtensions$kotlin_gradle_plugin_common'.
   > Could not resolve all dependencies for detachedConfiguration5.
      > Could not find org.springframework.kafka:spring-kafka-bom:3.1.1.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/org/springframework/kafka/spring-kafka-bom/3.1.1/spring-kafka-bom-3.1.1.pom
          - https://repo.spring.io/milestone/org/springframework/kafka/spring-kafka-bom/3.1.1/spring-kafka-bom-3.1.1.pom
          - https://repo.spring.io/snapshot/org/springframework/kafka/spring-kafka-bom/3.1.1/spring-kafka-bom-3.1.1.pom
        Required by:
            project :spring-integration-test-support
```

### Expected behavior

Our expectation that `jf gradle` together with its `build-info-extractor-gradle` would populate some Gradle init script to include that provided `jf gradlec --repo-resolve libs-staging-local` with respective credentials extracted from the `JF_ENV_SPRING` secret.

For now our workaround to not use JFrog CLI at all, but rather manually generate Gradle init script with some Artifactory exposed credentials:
```
      - name: Prepare Samples project against Staging
        run: |
          printf "allprojects {
                  repositories {
                    maven { 
                      url 'https://repo.spring.io/libs-staging-local'
                      credentials {
                        username = '$ARTIFACTORY_USERNAME'
                        password = '$ARTIFACTORY_PASSWORD'
                      }
                    }
                  }
                }" > staging-repo-init.gradle
          
          sed -i "1,/springIntegrationVersion.*/s/springIntegrationVersion.*/springIntegrationVersion='${{ inputs.releaseVersion }}'/" build.gradle  

      - name: Verify Spring Integration Samples against staged release
        run: gradle check --init-script staging-repo-init.gradle
```

### JFrog CLI version

2.50.4

### Operating system type and version

runs-on: ubuntu-latest

**Comment by yahavi** [comment]
@artembilan 
Thanks for reporting this issue.
We fixed it in version [2.52.3](https://github.com/jfrog/jfrog-cli/releases/tag/v2.52.3) of the JFrog CLI by https://github.com/jfrog/artifactory-gradle-plugin/pull/90.

Feel free to upgrade. We'd appreciate your feedback on that!

**Comment by tkatrichenko** [comment]
> We fixed it in version [2.52.3](https://github.com/jfrog/jfrog-cli/releases/tag/v2.52.3) of the JFrog CLI by https://github.com/jfrog/artifactory-gradle-plugin/pull/90.

Hello, this fix just broke my builds. 
I configured gradle in CI via `jf gradle-config --server-id-resolve my-server --server-id-deploy my-server --repo-resolve gradle-local --repo-deploy gradle-local`

I got `Build was configured to prefer settings repositories over project repositories but repository 'MavenRepo' was added by build file 'build.gradle'` and build failed

So I have to remove server-id-resolve and server-id-deploy from the command. Now it works

Think it confirms that this bug is resolved

**Comment by artembilan** [comment]
I can confirm that it works now as expected.
So, we have this workflow config:
```
      - uses: jfrog/setup-jfrog-cli@v3
        env:
          JF_ENV_SPRING: ${{ secrets.JF_ARTIFACTORY_SPRING }}

      - name: Configure JFrog Cli
        run: jf gradlec --repo-resolve libs-release-staging

      - name: Verify Spring Integration Kafka against staged release
        run: |
          sed -i "1,/springKafkaVersion.*/s/springKafkaVersion.*/springKafkaVersion='${{ inputs.releaseVersion }}'/" build.gradle
          jf gradle :spring-integration-kafka:check
```
And logs demonstrate the proper attempt to resolve an artifact:
```
       > Could not find org.springframework.kafka:spring-kafka-bom:3.1.2.
        Searched in the following locations:
          - https://repo.spring.io/artifactory/libs-release-staging/org/springframework/kafka/spring-kafka-bom/3.1.2/spring-kafka-bom-3.1.2.pom
```

Thank you!
